tools/clspv-opt/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+# Copyright 2017 The Clspv Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if(NOT ENABLE_CLSPV_OPT)
+  return()
+endif()
+
+add_executable(clspv-opt ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
+
+if(APPLE)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
+    -DLLVM_OPT=\\\"${CLSPV_BINARY_DIR}/third_party/llvm/bin/opt\\\" \
```

Comment:
I think this may be incorrect for external llvm builds.

Reply:
It was.  Also found that `lib/CMakeLists.txt` had a wrong path to clang when doing out-of-tree builds.  Fixed that too.